docs/proposals/config_data.md

```diff
@@ -0,0 +1,298 @@
+
+### Environment Variables
+
+The component configuration workflow is given below:
+
+1. ``ConfigData`` object is created through the API server.
+2. New components using the ``ConfigData`` object are created.
+3. ``ConfigData`` object is retrieved using the downward API.
+4. ``Data`` key/value pairs of ``ConfigData`` will be exposed as environment variables (``EnvVarSource``).
+5. API server is watched for modifications to the object.
+6. After a modification, actions 3, 4, and 5 are repeated.
+
+### Mounted Volume
+
+The component configuration workflow is given below:
+
+1. ``ConfigData`` object is created through the API server.
+2. New components using the ``ConfigData`` object are created.
+3. ``ConfigData`` object is retrieved using the downward API.
+4. ``ConfigData`` will be mounted as a new volume (``ConfigDataVolumeSource``) where keys are file names and values are file contents.
+5. API server is watched for modifications to the object.
+6. After a modification, actions 3, 4, and 5 are repeated.
+
+## Examples
+
+#### Consuming ConfigData as mounted volumes
+
+``pod-config-volume`` is intended to be used as a mounted volume containing two files with their respective contents.
+```json
+{
+    "apiVersion": "extensions/v1beta1",
+    "metadata": {
+        "name": "pod-config-volume",
+    },
+    "kind": "ConfigData",
+    "data": {
+        "component_config": "COMPONENT_CONFIG_STRING_CONTENTS",
+        "component_metadata": "COMPONENT_METADATA_STRING_CONTENTS"
+    }
+}
+```
+
+Then, configuration objects are persisted to the API server as follows:
+```bash
+kubectl create -f pod-configuration-vol.json 
+```
+
+ExampleReplicationController starts 2 replicas that runs the `busybox` container.
+In this use case, the ``ConfigData`` object will be mounted as a volume and contain
+two files named "config.yaml" and "metadata.json".
+```yaml
+apiVersion: extensions/v1beta1
+kind: ReplicationController
+metadata:
+  name: ExampleReplicationController
+spec:
+  replicas: 2
+  selector:
+    app: busybox
+  template:
+    metadata:
+      labels:
+        app: busybox
+    spec:
+      containers:
+        - name: busybox
+          image: gcr.io/google_containers/busybox
+          ports:
+            - containerPort: 80
+          volumeMounts:
+            - name: config-data-volume
+      volumes:
+      - name: config-data-volume
+        configDataVolumes:
+          items:
+            - path: "config.yaml"
+              configDataRef:
+                from: pod-config-volume
+                fieldPath: component_config
+            - path: "metadata.json"
+              configDataRef:
+                from: pod-config-volume
+                fieldPath: component_metadata
+```
+
+Now, we create the replication controller:
+```bash
+kubectl create -f example-replication-controller.yaml
+```
+
+New pods created by the ExampleReplicationController will use the ``pod-config-volume`` objects when starting/replicating pods.
```

Comment:
This sentence reads a little awkwardly and I'm not sure what it adds -- maybe we can strike it.